mapie/conformity_scores/residual_conformity_scores.py

```diff
@@ -132,3 +141,275 @@ def get_estimation_distribution(
+        split_size: Optional[Union[int, float]] = None,
+        random_state: Optional[Union[int, np.random.RandomState]] = None
+    ) -> None:
+        super().__init__(sym=True, consistency_check=False)
+        self.prefit = prefit
+        self.residual_estimator = residual_estimator
+        self.split_size = split_size
+        self.random_state = random_state
+
+    def _check_estimator(
+        self, estimator: Optional[RegressorMixin] = None
+    ) -> RegressorMixin:
+        """
+        Check if estimator is ``None``,
+        and returns a ``LinearRegression`` instance if necessary.
+        If the ``prefit`` attribute is ``True``,
+        check if estimator is indeed already fitted.
+
+        Parameters
+        ----------
+        estimator: Optional[RegressorMixin]
+            Estimator to check, by default ``None``.
+
+        Returns
+        -------
+        RegressorMixin
+            The estimator itself or a default ``LinearRegression`` instance.
+
+        Raises
+        ------
+        ValueError
+            If the estimator is not ``None``
+            and has no ``fit`` nor ``predict`` methods.
+
+        NotFittedError
+            If the estimator is not fitted
+            and ``prefit`` attribute is ``True``.
+        """
+        if estimator is None:
+            return LinearRegression()
+        else:
+            if not (hasattr(estimator, "fit") and
+                    hasattr(estimator, "predict")):
+                raise ValueError(
+                    "Invalid estimator. "
+                    "Please provide a regressor with fit and predict methods."
+                )
+            if self.prefit:
+                if isinstance(estimator, Pipeline):
+                    check_is_fitted(estimator[-1])
+                else:
+                    check_is_fitted(estimator)
+            return estimator
+
+    def _check_parameters(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        y_pred: ArrayLike
+    ) -> Tuple[NDArray, NDArray, NDArray, RegressorMixin]:
+        """
+        Checks all the parameters of the class. Raises an error if the
+        parameter are not well defined.
+
+        Parameters
+        ----------
+        X: ArrayLike
+            Observed values.
+
+        y: ArrayLike
+            Target values.
+
+        y_pred: ArrayLike
+            Predicted targets.
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray, NDArray, RegressorMixin]
+        Well initiated and typed :
+            - X
+            - y
+            - y_pred
+            - residual_estimator
+        """
+        residual_estimator = self._check_estimator(
+            self.residual_estimator
+        )
+        check_random_state(self.random_state)
+        if self.split_size is None:
+            self.split_size = 0.5
+        X, y, y_pred = indexable(X, y, y_pred)
+        X = np.array(X)
+        y = np.array(y)
+        y_pred = np.array(y_pred)
```

Comment:
```suggestion
        X = cast(NDArray, X)
        y = cast(NDArray, y)
        y_pred = cast(NDArray, y_pred)
```
To access to `cast`: `from typing import cast`

Reply:
Because of how it is used (especially to get certain group of indexes in `get_signed_conformity_score`, we need to keep `np.array` instead of `cast(NDArray, ..)`

Reply:
I don't understand why. In other files, we use `cast` (see https://github.com/scikit-learn-contrib/MAPIE/blob/master/mapie/regression/regression.py#L385). Why is it different here?

Reply:
Because here it is used with specific numpy function, for example :
`np.argwhere(
            np.logical_not(np.isnan(y_pred))
        ).reshape((-1,))`

Reply:
Of course I tested it with `cast` and it generates an error

Reply:
That's why I changed the usual way of MAPIE :)

Reply:
That's the information I was missing, thanks for the clarification :)
And do you know why it does not work?